baseband/vdif/header.py

```diff
@@ -28,6 +32,41 @@
                   precision=9)
 
 
+_vdif_edv_header_classes = {}
+"""Dict for storing pairs of VDIF EDV and header classes."""
+
+class _VDIFHeaderRegistry(type):
+    """
+    Registry of VDIF Header EDV types, using the ``_vdif_edv_header_classes``
+    dict.  Checks for keyword and subclass conflicts before registering.
+    """
+    _vdif_edv_header_classes = _vdif_edv_header_classes
+
+    def __init__(cls, name, bases, dct):
+
+        # Check that class is subclass of VDIFHeader
+        if not issubclass(cls, VDIFHeader):
+            raise TypeError(cls.__name__ + " is not a subclass of VDIFHeader!")
+
+        # Extract edv from class; convert to -1 if edv == False (VDIFLegacy)
+        edv = cls._edv
+        if edv is False:
+            edv = -1
+
+        # Check if EDV is already registered (edv == None is invalid).
+        if edv is None:
+            raise ValueError("edv cannot be None!")
```

Comment:
This basically means someone did not override `_edv` on their subclass. Maybe make the error message more informative:
E.g., "edv cannot be None. It should be overriden by the subclass".

Note: you need to add an exception here for intermediate headers (see comment below).